frontend/app/src/pages/playerSelection/PlayerSelection.tsx

```diff
@@ -1,49 +1,139 @@
+import bluePlayerIcon from "@/assets/bluePlayerIcon.svg";
+import redPlayerIcon from "@/assets/redPlayerIcon.svg";
 import { RequireLogout } from "@/components/requireLogout/RequireLogout";
-import { Button } from "@/components/ui/button.tsx";
-import { useDispatchUser } from "@/services/userContext/UserContextHelpers.ts";
+import { GameState } from "@/domain/GameState";
+import { Lobby } from "@/domain/Lobby";
+import { useToast } from "@/hooks/use-toast";
+import {
+	useDispatchLobby,
+	useLobby,
+} from "@/services/lobbyContext/LobbyContextHelpers";
+import { createLobby, joinLobby } from "@/services/rest/LobbyRest";
+import {
+	useDispatchUser,
+	useUser,
+} from "@/services/userContext/UserContextHelpers.ts";
+import { useEffect, useState } from "react";
 import { useNavigate } from "react-router-dom";
-import { useState } from "react";
+import styles from "./PlayerSelection.module.css";
 
 function PlayerSelection() {
 	const navigate = useNavigate();
-	const dispatcher = useDispatchUser();
+
+	const userDispatcher = useDispatchUser();
+	const lobbyDispatch = useDispatchLobby();
+
+	const lobby = useLobby();
+	const user = useUser();
+
 	const [blockLogout, setBlockLogout] = useState<boolean>(false);
+	const [shouldRedirect, setShouldRedirect] = useState(false);
+
+	const { toast } = useToast();
+
+	useEffect(() => {
+		if (lobby && user && shouldRedirect) {
+			navigate("/player-coding");
+			setBlockLogout(false);
+		}
+	}, [lobby, user, shouldRedirect, navigate]);
+
+	useEffect(() => {
+		async function joinLobbyWithPlayer(userName: string, lobbyId: number) {
+			if (!lobbyDispatch) {
+				return;
+			}
+			joinLobby(userName, lobbyId)
+				.catch(handleLobbyError)
+				.then(() => {
+					if (!lobby || lobby.id !== lobbyId) {
+						const newLobby = new Lobby(
+							lobbyId,
+							[userName],
+							GameState.NOT_STARTED,
+						);
+						lobbyDispatch({ type: "join", lobby: newLobby });
+					}
+					setShouldRedirect(true);
+				});
+		}
 
-	function setPlayerAndRedirectToCoding(
-		actionType: "setPlayerA" | "setPlayerB",
-	) {
-		if (dispatcher) {
+		function handleLobbyError() {
+			toast({
+				title: "Oopsie👉👈",
+				description: "Something went wrong while joining the lobby 😢",
+				variant: "destructive",
+			});
+			if (userDispatcher) userDispatcher({ type: "logout", user: null });
+			navigate("/lobby-selection");
+			setBlockLogout(false);
+		}
+
+		if (!user || !lobbyDispatch) {
+			return;
+		} else if (lobby) {
+			joinLobbyWithPlayer(user.name, lobby.id);
+		} else {
+			createLobby()
+				.catch(handleLobbyError)
+				.then((lobbyID) => {
+					if (lobbyID) joinLobbyWithPlayer(user.name, lobbyID);
+				});
+		}
+	}, [user, lobby, lobbyDispatch, toast, userDispatcher, navigate]);
```

Comment:
does this have to be a single useEffectHook? Seems very long :/

Reply:
the logic of this one is actually not that long (the if-else at the end) and if i would put the two functions outside of the useEffect i would get ESLINT warnings saying to update the dependency array. (the official fix for this is to declare the function inside the useEffect hooks : [see here](https://github.com/facebook/react/issues/14920)

Reply:
but i tried to reduce the visual complexity a bit by reducing the use of .then(....) and using await statements instead